coordinator/src/main/scala/com/socrata/datacoordinator/resources/collocation/CollocationSodaResource.scala

```diff
@@ -47,23 +48,34 @@ abstract class CollocationSodaResource extends SodaResource {
   def datasetNotFound(datasetInternalName: DatasetInternalName, resp: HttpResponse = NotFound): HttpResponse =
     errorResponse(resp, CollocationError.DATASET_DOES_NOT_EXIST, "dataset" -> JString(datasetInternalName.underlying))
 
-  def withBooleanParam(name: String, req: HttpRequest)(handleRequest: Boolean => HttpResponse): HttpResponse = {
-    val param = Option(req.servletRequest.getParameter(name)).getOrElse("false")
+  def withTypedParam[T](name: String, req: HttpRequest, defaultValue: T)(handleRequest: T => HttpResponse): HttpResponse = {
+    val param = Option(req.servletRequest.getParameter(name)).getOrElse(defaultValue.toString)
     try {
-      handleRequest(param.toBoolean)
+      val parsedParam = defaultValue match {
```

Comment:
might be nice to have this function instead take a function from `String => T` so it can be generic without this match case.